### ISSUE TITLE
dind: failed to adjust OOM score for shim

### ISSUE THREAD

**Issue by strowi** [k3s-io/k3s#1016]
Hi,

i'm trying to run k3s in kubernetes with our gitlab-pipline, but when i try to deploy a simple nginx, i get the following error:

```
kubectl -n $KUBE_NAMESPACE get ev
LAST SEEN   TYPE      REASON                   OBJECT                                   MESSAGE
...
<unknown>   Normal    Scheduled                pod/nginx-deployment-85ff79dd56-hf4w2    Successfully assigned sys-ci-examples-k3s/nginx-deployment-85ff79dd56-hf4w2 to runner-mg4uz-es-project-378-concurrent-0vlcq8
14s         Warning   FailedCreatePodSandBox   pod/nginx-deployment-85ff79dd56-hf4w2    Failed create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: io.containerd.runc.v2: failed to adjust OOM score for shim: set shim OOM score: write /proc/797/oom_score_adj: invalid argument
: exit status 1: unknown
```

This is the pipeline config:
```

k3s:test:
  image: docker:stable
  variables:
    KUBE_NAMESPACE: $CI_PROJECT_PATH_SLUG
  script:
    - apk add wget
    - download_k3s
    - start_k3s
    - kubectl cluster-info
    - ensure_namespace
    - kubectl -n $KUBE_NAMESPACE apply -f manifests
    - sleep 30s
    - kubectl -n $KUBE_NAMESPACE get ev
    - kubectl -n $KUBE_NAMESPACE get all
    - get_pod nginx

before_script:
  - |

    function ensure_namespace() {
      echoinfo "Ensuring the ${KUBE_NAMESPACE} namespace exists..." true

      kubectl describe namespace "$KUBE_NAMESPACE" || kubectl create namespace "$KUBE_NAMESPACE"

      # create registry secret
      kubectl create secret -n $KUBE_NAMESPACE \
        docker-registry "$CI_REGISTRY" \
        --docker-server="$CI_REGISTRY" \
        --docker-username=gitlab-ci-token \
        --docker-password="$CI_BUILD_TOKEN" \
        --docker-email="$GITLAB_USER_EMAIL" \
        -o yaml --dry-run | kubectl apply -n $KUBE_NAMESPACE -f -
    }

    function download_k3s() {
      wget -q https://github.com/rancher/k3s/releases/download/v0.7.0/k3s -O /usr/bin/k3s
      chmod +x /usr/bin/k3s
    }

    function start_k3s() {

      k3s server > k3s.log 2>&1 &

      until [ -f /etc/rancher/k3s/k3s.yaml ]; do
        sleep 1
        echo -n .
      done

      export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
      kubectl version
    }

```

It seems k3s thinks its running into an OOM, which can't be true, because there are no limits on the pod or namespace running it...?

```
kubectl describe no
Name:               runner-mg4uz-es-project-378-concurrent-0wqtdv
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=runner-mg4uz-es-project-378-concurrent-0wqtdv
                    kubernetes.io/os=linux
Annotations:        volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Tue, 05 Nov 2019 17:40:16 +0000
Taints:             node.kubernetes.io/not-ready:NoSchedule
Unschedulable:      false
Conditions:
  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----             ------  -----------------                 ------------------                ------                       -------
  MemoryPressure   False   Tue, 05 Nov 2019 17:40:16 +0000   Tue, 05 Nov 2019 17:40:16 +0000   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure     False   Tue, 05 Nov 2019 17:40:16 +0000   Tue, 05 Nov 2019 17:40:16 +0000   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure      False   Tue, 05 Nov 2019 17:40:16 +0000   Tue, 05 Nov 2019 17:40:16 +0000   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready            True    Tue, 05 Nov 2019 17:40:16 +0000   Tue, 05 Nov 2019 17:40:16 +0000   KubeletReady                 kubelet is posting ready status
Addresses:
  InternalIP:  10.42.8.221
  Hostname:    runner-mg4uz-es-project-378-concurrent-0wqtdv
Capacity:
 cpu:                24
 ephemeral-storage:  228182444Ki
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             65856924Ki
 pods:               110
Allocatable:
 cpu:                24
 ephemeral-storage:  221975881350
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             65856924Ki
 pods:               110
System Info:
 Machine ID:                 
 System UUID:                00000000-0000-0000-0000-0cc47a6aef68
 Boot ID:                    a472ac9f-809a-4be5-b752-be3cb9e7d854
 Kernel Version:             5.0.0-32-generic
 OS Image:                   Alpine Linux v3.10
 Operating System:           linux
 Architecture:               amd64
 Container Runtime Version:  containerd://1.2.5+unknown
 Kubelet Version:            v1.14.1-k3s.4
 Kube-Proxy Version:         v1.14.1-k3s.4
Non-terminated Pods:         (0 in total)
  Namespace                  Name    CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                  ----    ------------  ----------  ---------------  -------------  ---
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests  Limits
  --------           --------  ------
  cpu                0 (0%)    0 (0%)
  memory             0 (0%)    0 (0%)
  ephemeral-storage  0 (0%)    0 (0%)
Events:
  Type     Reason                   Age              From                                                    Message
  ----     ------                   ----             ----                                                    -------
  Normal   NodeHasSufficientMemory  0s (x2 over 0s)  kubelet, runner-mg4uz-es-project-378-concurrent-0wqtdv  Node runner-mg4uz-es-project-378-concurrent-0wqtdv status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure    0s (x2 over 0s)  kubelet, runner-mg4uz-es-project-378-concurrent-0wqtdv  Node runner-mg4uz-es-project-378-concurrent-0wqtdv status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID     0s               kubelet, runner-mg4uz-es-project-378-concurrent-0wqtdv  Node runner-mg4uz-es-project-378-concurrent-0wqtdv status is now: NodeHasSufficientPID
  Warning  SystemOOM                0s               kubelet, runner-mg4uz-es-project-378-concurrent-0wqtdv  System OOM encountered
  Normal   NodeAllocatableEnforced  0s               kubelet, runner-mg4uz-es-project-378-concurrent-0wqtdv  Updated Node Allocatable limit across pods
 
```

Not sure where i should look for the error, docker, containerd or kubernetes?

regards,
strowi

**Comment by dxas90** [comment]
It is  [containerd](https://github.com/containerd/containerd/issues/4837)

**Comment by viceice** [comment]
I think this is fixed in containerd v1.4.4, so all k3s versions from [`v1.19.9`](https://github.com/k3s-io/k3s/releases/tag/v1.19.9%2Bk3s1) are fixed

**Comment by brandond** [comment]
Closing as resolved by upstream.
